### ISSUE TITLE
Normal::Inverse timing out

### ISSUE THREAD

**Issue by gente002** [markrogoyski/math-php#264]
When seeding my database with the Normal::Inverse function, I came across an oddity.  If I run the bottom command my seed stalls.  However, when I add or subtract 1 digit from the first parameter, it works perfectly fine.  Any reason as to why this issue persists only in the bottom case?

Right now my work around is rounding to 5 digits for the first parameter but I want to understand this so I can implement a proper fix.

**Commands**
```
Continuous\Normal::inverse(0.49999999995,0,1);    //Works as expected
Continuous\Normal::inverse(0.499999995,0,1);      //Works as expected
Continuous\Normal::inverse(0.4999999995,0,1);     //Stalls out
```

**Comment by markrogoyski** [comment]
Hi @gente002,

I was able to reproduce the issue on the latest version. I'll take a closer look to see what might be going on. Thanks for reporting and providing data the reproduces the issue.

Btw, it looks like you may be using an older version of the library? The probability distributions are now instantiable objects with object methods.

@Beakerboy 
Just taking a quick look, it seems like the diff is swinging back and forth between two really small numbers while the CDF does the same, so it never quite decreases enough to go below the tolerance. I wonder if there should be a check on the running guess and iterations as well, and at some really small tolerance and x number of iterations just consider it zero. Thoughts?

**Comment by Beakerboy** [comment]
This is off the top of my head...the code is not in front of me. I wonder if the function could retain the previous two values. If it is flipping between two, split the difference and continue. If it’s flipping between two values, we know the real root is in between them somewhere.

**Comment by markrogoyski** [comment]
Hi @gente002,

I've added a fix to handle this edge case when computing the inverse of a probability distribution. The fix is currently in the _develop_ branch. If you checkout or composer-require the develop branch and install, you should be able to test the fix. I've added unit tests cases for the examples you provided. If there are any other cases where you know this was happening, please provide them and I'll add them to the unit test regression suite.

Please test the fix and verify the inverse works as you expect. This fix will be in the next release.

@Beakerboy,

The fix is in the develop branch in commit https://github.com/markrogoyski/math-php/commit/085469a0107732fba39d402f7a44b44852fb8a37.

Take a look and see if you have any concerns with this approach. Thanks.

**Comment by gente002** [comment]
Thanks @markrogoyski!  I will have a look at this in the next few days and let you know if I have any issues.  I have never required a different branch for testing purposes so a bit of Googling is necessary.

**Comment by gente002** [comment]
I took a look at this this morning and everything worked as expected.  Thanks for looking in to this @markrogoyski  and @Beakerboy.

**Comment by markrogoyski** [comment]
This has been implement and released in version 0.40.0. Thank you for reporting and your continued interest in MathPHP.
